src/lib/temple/front/chain.ts

```diff
@@ -40,9 +41,12 @@ function useNewBlockTriggers() {
   };
 }
 
-export function useOnBlock(callback: (blockHash: string) => void) {
-  const tezos = useTezos();
+export function useOnBlock(callback: (blockHash: string) => void, altTezos?: TezosToolkit) {
+  const currentTezos = useTezos();
   const blockHashRef = useRef<string>();
+  const callbackRef = useCallbackRef(callback);
```

Comment:
Should we update this value like here: https://github.com/madfish-solutions/templewallet-extension/pull/1074/files#diff-388fa68100876e550afeaab276b9c46095d595fb18eae699affe241308db6ec6R26 ?